command/agent/agent.go

```diff
@@ -314,9 +314,18 @@ func (a *Agent) clientConfig() (*clientconfig.Config, error) {
 	if conf == nil {
 		conf = clientconfig.DefaultConfig()
 	}
+
+	// If we are running a server, append both its bind and advertise address so
+	// we are able to at least talk to the local server even if that isn't
+	// configured explicitly. This handles both running server and client on one
+	// host and -dev mode.
+	conf.Servers = a.config.Client.Servers
 	if a.server != nil {
-		conf.RPCHandler = a.server
```

Comment:
Why was setting the RPCHandler removed? What replaces this functionality?

Reply:
If you read my description you will see why this is problematic. By using in memory only RPCs the mechanism to track client connections doesn't work and thus proxied connections break. Therefore we switch to using the normal mechanism of RPCs and just explicitly seed the client's server list with the address the server is bound to.